pd/src/pending_block.rs

```diff
@@ -112,6 +134,13 @@ impl PendingBlock {
             transaction_id: [0; 32],
         };
 
+        self.add_note(commitment, note_data);
+
+        self.reward_counter += 1;
+    }
+
+    /// Adds a new note to this pending block.
+    pub fn add_note(&mut self, commitment: note::Commitment, data: NoteData) {
```

Comment:
The same code was being used to add a note in `add_transaction` and `add_validator_reward_note`; we also need to call it directly to add the unbonding notes to the `PendingBlock` when they unbond. I've factored it out into a function here.